OsmAnd/src/net/osmand/plus/views/mapwidgets/widgetstates/ExternalSensorWidgetState.java

```diff
@@ -0,0 +1,72 @@
+package net.osmand.plus.views.mapwidgets.widgetstates;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.plugins.externalsensors.devices.sensors.SensorWidgetDataFieldType;
+import net.osmand.plus.settings.backend.ApplicationMode;
+import net.osmand.plus.settings.backend.preferences.OsmandPreference;
+import net.osmand.plus.settings.enums.ExternalDeviceShowMode;
+import net.osmand.util.Algorithms;
+
+public class ExternalSensorWidgetState extends WidgetState {
+
+	public static final String SHOW_MODE_ID = "show_mode";
+
+	private final SensorWidgetDataFieldType widgetType;
+	private final OsmandPreference<ExternalDeviceShowMode> showModePreference;
+
+
+	public ExternalSensorWidgetState(@NonNull OsmandApplication app, @Nullable String customId, SensorWidgetDataFieldType widgetType) {
+		super(app);
+		this.widgetType = widgetType;
+		this.showModePreference = registerShowModePreference(customId);
+	}
+
+	@NonNull
+	public OsmandPreference<ExternalDeviceShowMode> getShowModePreference() {
+		return showModePreference;
+	}
+
+	@NonNull
+	@Override
+	public String getTitle() {
+		return app.getString(widgetType.getWidgetType().titleId);
+	}
+
+	@Override
+	public int getSettingsIconId(boolean nightMode) {
+		return nightMode ? widgetType.nightIconId : widgetType.dayIconId;
+	}
+
+	@Override
+	public void changeToNextState() {
+		ExternalDeviceShowMode currentMode = showModePreference.get();
+		if (currentMode.ordinal() == ExternalDeviceShowMode.values().length - 1) {
+			showModePreference.set(ExternalDeviceShowMode.values()[0]);
+		} else {
+			showModePreference.set(ExternalDeviceShowMode.values()[currentMode.ordinal() + 1]);
+		}
```

Comment:
Could we add something simpler like [ZoomLevelType.next()](https://github.com/osmandapp/Osmand/blob/dd53b92786c4818e9376ef2e1bb3c1bddcd0591c/OsmAnd/src/net/osmand/plus/views/mapwidgets/widgetstates/ZoomLevelWidgetState.java#L81-L84)?